common/scala/src/whisk/core/entity/WhiskTrigger.scala

```diff
@@ -60,12 +100,28 @@ case class WhiskTrigger(
     limits: TriggerLimits = TriggerLimits(),
     version: SemVer = SemVer(),
     publish: Boolean = false,
-    annotations: Parameters = Parameters())
+    annotations: Parameters = Parameters(),
+    rules: Option[Map[Namespace, ReducedRule]] = Some(Map[Namespace, ReducedRule]()))
     extends WhiskEntity(name) {
 
     require(limits != null, "limits undefined")
 
     def toJson = WhiskTrigger.serdes.write(this).asJsObject
+
+    def withoutRules = WhiskTriggerResponse(namespace, name, parameters, limits, version, publish, annotations)
+
+    def addRule(rulename: Namespace, rule: ReducedRule) = {
```

Comment:
add javadoc, clarify that rulename is fully qualified via namespace. same for remove rule below.